lib/trino-record-decoder/src/main/java/io/trino/decoder/DispatchingRowDecoderFactory.java

```diff
@@ -31,9 +32,9 @@ public DispatchingRowDecoderFactory(Map<String, RowDecoderFactory> factories)
         this.factories = ImmutableMap.copyOf(factories);
     }
 
-    public RowDecoder create(String dataFormat, Map<String, String> decoderParams, Set<DecoderColumnHandle> columns)
+    public RowDecoder create(String dataFormat, Map<String, String> decoderParams, Set<DecoderColumnHandle> columns, ConnectorSession session)
```

Comment:
Should session be the first parameter here too?

Reply:
Makes sense, thanks! We can extend the suggestion for encoder parameters and create a `DecoderRowSpec` - will add both as separate commits.